### PR TITLE
[OM] Add support for boolean type.

### PR DESCRIPTION
**Pull request by prithayan**
Add the boolean type `om.bool`, the boolean attribute and the corresponding lowering from `firrtl.bool`.

### CODE REVIEW

**Review by seldridge — APPROVED** [review]

**Review by dtzSiFive — APPROVED** [review]
LGTM, thanks!

File: include/circt/Dialect/OM/OMAttributes.td [hunk 1]
```diff
@@ -121,4 +121,18 @@ def OMPathAttribute : AttrDef<OMDialect, "Path", [TypedAttrInterface]> {
   }];
 }
 
+def OMBoolAttr : AttrDef<OMDialect, "Bool", [TypedAttrInterface]> {
+  let summary = "An attribute that represents a boolean value";
+
+  let mnemonic = "bool";
+
+  let parameters = (ins "mlir::BoolAttr":$value);
```
Comment (dtzSiFive) [inline include/circt/Dialect/OM/OMAttributes.td:129]: LGTM but curious the pros/cons here re:wrapping BoolAttr? But works for me!

**Comment by uenoku** [comment]
Is there reason we cannot use i1? om.integer seems reasonable to model big integer but om.bool looks redundant to me.

**Comment by dtzSiFive** [comment]
> Is there reason we cannot use i1? om.integer seems reasonable to model big integer but om.bool looks redundant to me.

As long as we can reliably differentiate between a Bool and an integer that happens to be 1-bit.

I think with the `om.integer` introduced this will be the case, so maybe it could be that way too.

**Comment by prithayan** [comment]
Closing this, as per our discussions, `BoolAttr` is enough to represent boolean constants.

**Pull request closed without merging**